Source/WTF/wtf/text/StringCommon.h

```diff
@@ -522,7 +522,17 @@ template<typename SearchCharacterType, typename MatchCharacterType>
     requires(TriviallyComparableCodeUnits<SearchCharacterType, MatchCharacterType>)
 size_t findIgnoringASCIICase(std::span<const SearchCharacterType> source, std::span<const MatchCharacterType> matchCharacters, size_t startOffset = 0)
 {
-    ASSERT(source.size() >= matchCharacters.size());
+    if (matchCharacters.empty())
+        return 0;
+
+    auto sum = checkedSum<size_t>(matchCharacters.size() + startOffset);
```

Comment:
Using the addition operator and _then_ passing the result to `checkedSum()` cannot check for overflow, we want to have both parameters added to the template function:

```suggestion
    auto sum = checkedSum<size_t>(matchCharacters.size(), startOffset);
```